nncf/tensorflow/pruning/filter_pruning/algorithm.py

```diff
@@ -240,6 +240,12 @@ def set_pruning_level(self, pruning_level: float,
         if run_batchnorm_adaptation:
             self._run_batchnorm_adaptation()
 
+    @property
+    def maximal_compression_rate(self) -> float:
+        max_flops, _ = self._calculate_flops_and_weights_in_uniformly_pruned_model(1.0)
+        max_compression_rate = 1 - max_flops / max(self.full_flops, 1)
+        return max_compression_rate
```

Comment:
1) `max_flops` is a bit confusing name because it is minimum possible flops for the model after the pruning. `min_flops` is clearer in my opinion
2) `max(self.full_flops, 1)` is needed for a case when a model has 0 flops, right?

Reply:
1. I see. If I change, it will be inconsistent with this part [here](https://github.com/openvinotoolkit/nncf/blob/develop/nncf/tensorflow/pruning/filter_pruning/algorithm.py#L143) . Should it be changed it to `min_` in both places?
2. Yes

Reply:
1) Yes
2) 👍

Reply:
Following that trail, decided to put maximal compression rate computation inside existing `PrunedModelTheoreticalBorderline` where it seems to belong